relay-sampling/src/dsc.rs

```diff
@@ -37,7 +37,7 @@ pub struct DynamicSamplingContext {
     /// transaction.
     ///
     /// Set on transaction start, or via `scope.transaction`.
-    #[serde(default)]
+    #[serde(default, alias = "segment_name")]
```

Comment:
The docs also need updating.

Maybe we should the alias and the field name or we have a better field name which describes on a higher level what we store here?